components/feature/search/src/main/java/mozilla/components/feature/search/SearchFeature.kt

```diff
@@ -0,0 +1,57 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.feature.search
+
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.FlowPreview
+import kotlinx.coroutines.cancel
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.flow.distinctUntilChangedBy
+import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.flow.mapNotNull
+import mozilla.components.browser.state.action.ContentAction
+import mozilla.components.browser.state.selector.selectedTab
+import mozilla.components.browser.state.store.BrowserStore
+import mozilla.components.concept.engine.search.SearchRequest
+import mozilla.components.lib.state.ext.flowScoped
+import mozilla.components.support.base.feature.LifecycleAwareFeature
+import mozilla.components.support.utils.ext.toNullablePair
+
+/**
+ * Lifecycle aware feature that forwards [SearchRequest]s from the [store] to [performSearch], and
+ * then consumes them.
+ *
+ * NOTE: that this only consumes SearchRequests, and will not hook up the [store] to a source of
+ * SearchRequests. See [SelectionActionDelegate] for use in generating SearchRequests.
+ */
+class SearchFeature(
+    private val store: BrowserStore,
+    private val performSearch: (SearchRequest) -> Unit
+) : LifecycleAwareFeature {
+    private var scope: CoroutineScope? = null
+
+    @FlowPreview
```

Comment:
We usually do this instead: https://github.com/mozilla-mobile/android-components/blob/master/components/support/webextensions/build.gradle#L29

Reply:
Oh neat!